File: scripts/solar/util/math/CodecUtil.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 2,["7"] = 2,["8"] = 2,["10"] = 2,["11"] = 9,["12"] = 10,["13"] = 11,["14"] = 12,["15"] = 9,["16"] = 19,["17"] = 20,["18"] = 21,["19"] = 22,["20"] = 22,["21"] = 22,["22"] = 22,["23"] = 19,["24"] = 28,["25"] = 29,["26"] = 30,["27"] = 31,["28"] = 28,["29"] = 37,["30"] = 38,["31"] = 39,["32"] = 40,["33"] = 37,["34"] = 46,["35"] = 47,["36"] = 48,["37"] = 46,["38"] = 2});
local ____exports = {}
____exports.default = __TS__Class()
local CodecUtil = ____exports.default
CodecUtil.name = "CodecUtil"
function CodecUtil.prototype.____constructor(self)
end
function CodecUtil.saesEncode(self, content, password)
    require("solar_aes")
    local fun = _G.saesEncode
    return ____exports.default:sBase64Encode(fun(content, password))
end
function CodecUtil.saesDecode(self, ciphertext, password)
    require("solar_aes")
    local fun = _G.saesDecode
    return fun(
        ____exports.default:sBase64Decode(ciphertext),
        password
    )
end
function CodecUtil.sBase64Encode(self, content)
    local base64 = require("base64")
    local fun = base64.encode
    return fun(content)
end
function CodecUtil.sBase64Decode(self, ciphertext)
    local base64 = require("base64")
    local fun = base64.decode
    return fun(ciphertext)
end
function CodecUtil.md5hex(self, content)
    local md5 = require("md5")
    return md5.sumhexa(content)
end
____exports.default = CodecUtil
return ____exports

File: src/solar/util/math/CodecUtil.ts
/** @noSelfInFile */
export default class CodecUtil {


    /**
     * @param content 需要加密的内容
     * @param password 密码
     */
    static saesEncode(content: string, password?: string): string {
        require("solar_aes")
        let fun: (content: string, password?: string) => string = _G.saesEncode;
        return CodecUtil.sBase64Encode(fun(content, password))
    }

    /**
     * @param ciphertext 密文
     * @param password 密码
     */
    static saesDecode(ciphertext: string, password?: string): string {
        require("solar_aes")
        let fun: (ciphertext: string, password?: string) => string = _G.saesDecode;
        return fun(CodecUtil.sBase64Decode(ciphertext), password)
    }

    /**
     * @param content 需要编码的内容
     */
    static sBase64Encode(content: string): string {
        let base64: NoSelf = require("base64")
        let fun: (content: string) => string = base64.encode;
        return fun(content)
    }

    /**
     * @param ciphertext 需要解码的内容
     */
    static sBase64Decode(ciphertext: string): string {
        let base64: NoSelf = require("base64")
        let fun: (ciphertext: string) => string = base64.decode;
        return fun(ciphertext)
    }

    /**
     * @param content 需要散列的内容
     */
    static md5hex(content: string): string {
        let md5: NoSelf = require("md5")
        return md5.sumhexa(content)
    }


}